cli/azd/internal/repository/app_init.go

```diff
@@ -234,6 +234,27 @@ func prjConfigFromDetect(
 			}
 		}
 
+		if prj.HasWebUIFramework() {
+			// By default, use 'dist'. This is common for frameworks such as:
+			// - TypeScript
+			// - Vue.js
+			svc.OutputPath = "dist"
+
+		loop:
+			for _, dep := range prj.Dependencies {
+				switch dep {
+				case appdetect.JsReact:
+					// react uses 'build'
+					svc.OutputPath = "build"
+					break loop
+				case appdetect.JsAngular:
+					// angular uses dist/<project name>
+					svc.OutputPath = "dist/" + filepath.Base(rel)
+					break loop
+				}
```

Comment:
Similar to my comment in another PR - this is good for now but lets start thinking on how we can data drive more of this through configuration vs code since we are primarily just codifying switch cases to modify config based on switch case decisions.

Reply:
While the current code does look like simple switch cases, this can evolve over time where imperative programming language constructs would be more useful to determine the final outcome. I want to avoid premature optimization if possible.

Codifying simple constructs is as a configuration is indeed a good idea, but when we need to.